Feature create api env

gonkunkun opened the pull request:
Rails6 + APIモードで起動するように設定

READMEに従ってdocker-composeコマンドを実行して、
「Yay! You’re on Rails!」が表示されること、Railsのバージョンがv6であることを確認して欲しいです

gonkunkun committed:
Railsの初期画面が起動することを確認

gonkunkun committed:
APIモード起動追加

gonkunkun committed:
rails6 ＋apiモードで起動するように変更

tachibanayu24 commented on api/src/.gitignore:
```diff
@@ -0,0 +1,31 @@
+/.bundle
+
+# Ignore the default SQLite database.
+/db/*.sqlite3
```
sqliteは使わないので、.gitignoreから外していいかなと思います。
.gitignoreは見直していいかも。

tachibanayu24 commented on api/src/Gemfile:
```diff
@@ -0,0 +1,62 @@
+# Use Puma as the app server
+gem 'puma', '~> 3.11'
+# Use SCSS for stylesheets
+gem 'sass-rails', '~> 5.0'
```
apiモードの場合、sassは使わないから要らない気がします。

tachibanayu24 commented on api/src/Gemfile:
```diff
@@ -0,0 +1,62 @@
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 5.2.3'
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3'
```
今回DBはmysqlで。

tachibanayu24 commented on api/src/Gemfile:
```diff
@@ -0,0 +1,62 @@
+ruby '2.5.6'
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 5.2.3'
```
今回railsは6で。

tachibanayu24 commented on api/src/.ruby-version:
```diff
@@ -0,0 +1 @@
+ruby-2.5.6
```
rubyは2.6が最近リリースされた気がするから、そっちを検討してもいいかも。
もちろん2.5.6でOKですが。

tachibanayu24 commented on api/src/app/views/layouts/application.html.erb:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
あれ、apiモードの場合、view以下って生成されるんでしたっけ。
apiモードでnewしていますか？

tachibanayu24 commented on etc/docker/api/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+# Railsコンテナ用Dockerfile
+
+# イメジのベースラインにRuby2.5.1を指定
```
イメジ -> イメージ typoですかね。

tachibanayu24 commented on etc/docker/api/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+# Railsコンテナ用Dockerfile
+
+# イメジのベースラインにRuby2.5.1を指定
+FROM ruby:2.5
```
dockerイメージがちょっと大きすぎる気が。
```
REPOSITORY                TAG                     IMAGE ID            CREATED             SIZE
shubun_api                latest                  d6cc54e9a72b        29 seconds ago      1.01GB
shubun_client             latest                  ed343cdd31bf        2 hours ago         18.2MB
shubun_builder            latest                  ad6cc19e2189        2 hours ago         290MB
```
alpineという軽量なリナックスディストリビューションをベースにしたdockerイメージを使うのはよくあるので、こちらの適用も検討してください。
> 参考
> https://hub.docker.com/_/ruby/?__hstc=233161921.7bf88e972fdd7a7debc8575bac5a80cf.1467849600079.1467849600080.1467849600081.1&__hssc=233161921.1.1467849600082&__hsfp=3105786926

tachibanayu24 commented on api/src/Gemfile:
```diff
@@ -1,5 +1,45 @@
-gem "rails" # コメントアウトを外す
\ No newline at end of file
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 6.0.0'
```
あ、別のコミットでrailsのバージョンとか変更していたんですね。

tachibanayu24 commented on api/src/README.md:
```diff
@@ -0,0 +1,24 @@
+# README
```
こういった不要ファイルは削除して、最小構成のapiディレクトリにしたほうが後々の開発が楽かもしれないです

tachibanayu24 commented on README.md:
````diff
@@ -6,6 +6,7 @@
 ## Setup
 
 ```
+$ docker-compose run api rails new --api .
````
これ、起動時にコンテナ内でrails newするんですか？
ローカルの内容をマウントして、起動時はrails server動かすとかにしたほうがいい気が。
docker-compose buildでイメージ作成して、docker-compose upでrails server起動。とか。

tachibanayu24 commented:
一通り確認しました。

> READMEに従ってdocker-composeコマンドを実行して、
> 「Yay! You’re on Rails!」が表示されること、Railsのバージョンがv6であることを確認して欲しいです

は確認できました。

ただ、叩いたらjsonで `health check ok` とか返ってくるような、簡単なapiエンドポイントをswaggerに定義して、それを実装する感じのところまでやったほうがいい気が。
不要ファイルの削除などもお願いしたいです。

tachibanayu24 commented on docker-compose.yml:
```diff
@@ -1,5 +1,20 @@
+  #   volumes:
+  #      - ./tmp/db:/var/lib/postgresql/data
+  api:
+    build:
```
`container_name: shubun_api`
とかほしいかも。
shubunの部分はあとで変えるとして。

tachibanayu24 commented on docker-compose.yml:
```diff
@@ -1,5 +1,20 @@
+    command: bash -c "rm -f tmp/pids/server.pid && bundle exec rails s -p 3031 -b '0.0.0.0'"
+    volumes:
+      - ./api/src:/app
+    ports:
```
どっちでもいいけど、どうせフォワーディングするからコンテナ内でのポートは3000でもいいのでは。

kawa-masaya committed:
swagger templete create

gonkunkun committed:
Merge pull request #14 from perches/feature/swagger-devenv

swagger templete create

gonkunkun committed:
以下を修正・追加
・apiコンテナ内のポートを3000に変更
・コンテナ名を設定
・「docker-compose run api rails new --api .」の手順を削除
・不要ファイル「README」を削除
・rubyのdockerイメージにalpineを使用し軽量化
・rubyのバージョンを2.6.4に変更

gonkunkun committed:
・docker-compose起動時に、db:migrate, setupを自動で実行するように entrypoint.shを修正
・CORS対応
・healthcheck用のSamplesコントローラ作成
